repo_resource/common.py

```diff
@@ -105,12 +105,24 @@ def getRevision(remote, remoteUrl, project, branch):
             if is_sha1(branch):
                 return (remote + '/' + project, branch)
             g = git.cmd.Git()
-            url, revision = (
+            url, revList = (
                 remote + '/' + project,
-                g.ls_remote(remoteUrl+'/'+project, branch).split()[0]
+                g.ls_remote(remoteUrl+'/'+project, branch).split()
             )
-            print('{}: {}'.format(url, revision))
-            return (url, revision)
+
+            # convert revision list to revision dict:
+            # ['SHA1', 'refs/heads/XXXX', 'SHA1', 'refs/heads/YYYY']
+            # -> {'refs/heads/XXXX': 'SHA1', 'refs/heads/YYYY': 'SHA1'}
```

Comment:
This code seems to be moved from the other commit.
Why is this now needed ?

Reply:
Well no this code is needed to fix the issue described in this commit.

It's not related to the other commits

Reply:
By converting the revision list to revision dict with `refs/heads/XXXX` as key, we ensure that the right revision is returned